client/src/main/java/io/pravega/client/batch/SegmentMetadata.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.batch;
+
+import com.google.common.annotations.Beta;
+import io.pravega.client.segment.impl.Segment;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+
+@Beta
+@Data
+@Builder
+public class SegmentMetadata {
```

Comment:
The term 'metadata' has another meaning in Pravega. Consider SegmentInputSplit which is familiar and seems to fit the situation.